aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -23,6 +23,17 @@
 namespace at {
 namespace native {
 
+namespace {
+torch::List<c10::optional<Tensor>> toListOfOptionalTensors(ArrayRef<Tensor> list) {
```

Comment:
I moved it to IndexingUtils.h since all current use cases are indexing related. Not super happy with it since it is probable that we'll need it for non-indexing-related stuff later on, open for better suggestions.